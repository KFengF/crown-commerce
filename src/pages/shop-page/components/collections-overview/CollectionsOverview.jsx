import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import CollectionPreview from '../collection-preview/CollectionPreview';
import { collectionsSelector } from '../../../../utils/redux/shop/shopSelectors';
import { CollectionsOverviewDiv } from './CollectionOverviewStyles';
/* import './CollectionsOverview.scss'; */

const CollectionsOverview = ({ collections }) => (
    <CollectionsOverviewDiv>
        { collections ? 
            Object.keys(collections).map(key => {
                const { id, ...otherCollectionAttrs } = collections[key];
                return <CollectionPreview 
                    key={ id } title={ key } 
                    { ...otherCollectionAttrs }
                />
            }) :
            null
        }
    </CollectionsOverviewDiv>
);

const mapState = createStructuredSelector({
    collections: collectionsSelector
});

export default connect(mapState)(CollectionsOverview);