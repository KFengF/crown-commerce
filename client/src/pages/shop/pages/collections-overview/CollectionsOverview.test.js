import React from 'react';
import { shallow } from 'enzyme';
import { CollectionsOverviewPage } from './CollectionsOverview.page';

describe('CollectionOverview page', () => {
    let mockCollections;
    let wrapper;

    beforeAll(() => {
        mockCollections = {
            hats: {
                id: 1,
                items: [
                    { name: 'hello', id: 1 },
                    { name: 'how', id: 2 },
                ]
            },
            sneakers: {
                id: 2,
                items: [
                    { name: 'are', id: 3 },
                    { name: 'you', id: 4 },
                ]
            }
        }
        const mockProps = {
            collections: mockCollections 
        }
        wrapper = shallow(<CollectionsOverviewPage { ...mockProps } />);
    });

    it('should render CollectionOverview page', () => {
        expect(wrapper).toMatchSnapshot();
    });

    describe('collections prop', () => {
        it('should be two CollectionPreview components', () => {
            expect(wrapper.find('CollectionPreview')).toHaveLength(2);
        });
    });
});