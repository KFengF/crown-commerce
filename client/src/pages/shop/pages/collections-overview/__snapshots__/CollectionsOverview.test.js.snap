// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionOverview page should render CollectionOverview page 1`] = `ShallowWrapper {}`;
